lib/Doctrine/ORM/Tools/Setup.php

```diff
@@ -126,19 +126,27 @@ public static function createConfiguration($isDevMode = false, $proxyDir = null,
         if ($isDevMode === false && $cache === null) {
             if (extension_loaded('apc')) {
                 $cache = new \Doctrine\Common\Cache\ApcCache();
-            } elseif (ini_get('xcache.cacher')) {
+            }
+            if ($cache === null && ini_get('xcache.cacher')) {
                 $cache = new \Doctrine\Common\Cache\XcacheCache();
-            } elseif (extension_loaded('memcache')) {
+            }
+            if ($cache === null && extension_loaded('memcache')) {
                 $memcache = new \Memcache();
-                $memcache->connect('127.0.0.1');
-                $cache = new \Doctrine\Common\Cache\MemcacheCache();
-                $cache->setMemcache($memcache);
-            } elseif (extension_loaded('redis')) {
+                $r = @$memcache->connect('127.0.0.1');
+                if ($r !== false) {
```

Comment:
This should throw if failed.

Reply:
It should not, it's a _dumb_ attempt to discover running memcache simply based on ext existence without configuration - if not found, try another storage.
@lcobucci suggested to drop Setup entirely in 3.0, sounds good to me.

Reply:
Yeah, probably a better solution. This is way too magic and simplistic.

Reply:
@Majkl578 could you be more specific about "drop Setup entirely"? You mean dropping the "connect" call, right?

Reply:
@ywarnier no, dropping the class entirely. It is completely unnecessary.